helm/flux-app/templates/source.yaml

```diff
@@ -11,6 +12,8 @@ metadata:
 ---
 {{- end }}
 {{- range .Values.sources }}
+{{- $is_gitrepository_crd := (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "gitrepositories.source.toolkit.fluxcd.io") -}}
+{{- if or $two_step_upgrade.unsupported $is_gitrepository_crd }}
```

Comment:
This basically says: if two-step installation is not supported, or there is CRDs already in place, render the CR.